atmo/coordinator/coordinator.go

```diff
@@ -144,70 +135,75 @@ func (c *Coordinator) SetupHandlers() *vk.Router {
 	return router
 }
 
-// CreateConnections establishes all of the connections described in the directive
+// CreateConnections establishes all of the connections described in the directive.
 func (c *Coordinator) createConnections() {
-	connections := c.App.Connections()
+	for _, application := range c.App.Applications() {
+		connections := c.App.Connections(application.Identifier, application.AppVersion)
 
-	if connections.NATS != nil {
-		address := rcap.AugmentedValFromEnv(connections.NATS.ServerAddress)
+		if connections.NATS != nil {
+			address := rcap.AugmentedValFromEnv(connections.NATS.ServerAddress)
 
-		gnats, err := nats.New(address)
-		if err != nil {
-			c.log.Error(errors.Wrap(err, "failed to nats.New for NATS connection"))
-		} else {
-			g := grav.New(
-				grav.UseLogger(c.log),
-				grav.UseTransport(gnats),
-			)
+			gnats, err := nats.New(address)
+			if err != nil {
+				c.log.Error(errors.Wrap(err, "failed to nats.New for NATS connection"))
+			} else {
+				g := grav.New(
+					grav.UseLogger(c.log),
+					grav.UseTransport(gnats),
+				)
 
-			c.connections[directive.InputSourceNATS] = g
+				c.connections[directive.InputSourceNATS] = g
```

Comment:
This could be problematic as multiple applications could have connections of the same type... we need a more specific key on this map (same for kafka down below)

Perhaps `{ident}.{version}.nats` or something?

Reply:
When storing them in the map we have access to the ident and version. There's only one place I found where it's being used:

https://github.com/suborbital/atmo/blob/issue-117/atmo/coordinator/coordinator_stream.go#L21-L30

There, as far as I can tell, we don't yet have access to ident or version. From what I can tell each `directive.Handler` encodes data that's in this part of a yaml file:
```yaml
handlers:
- type: request
  method: POST
  resource: /hello
  steps:
  - fn: helloworld
- type: request
  method: GET
  resource: /gabor
  steps:
  - fn: gabor
```

The coördinator has the appsource on it though. I think places like `streamConnectionForDirectiveHandler` should also try to get the connections for a specific app on the coördinator rather than ... a connection.

I'll rework this 🤔

Reply:
Fixed in https://github.com/suborbital/atmo/pull/121/commits/731fc85f00672b8dbfe65429617edbf856cb9703